src/EFCore/Query/QueryCompilationContext.cs

```diff
@@ -175,13 +178,24 @@ public virtual void AddTag(string tag)
             query,
             QueryContextParameter);
 
+        if (_queryExpressionInterceptor != null)
+        {
+            queryExecutorExpression = _queryExpressionInterceptor.CompilingQuery(
+                queryAndEventData.Query, queryExecutorExpression, queryAndEventData.EventData!);
```

Comment:
Shouldn't this be
```suggestion
                query, queryExecutorExpression, queryAndEventData.EventData!);
```
?

Reply:
It's actually intentionally the original LINQ expression, provided for information, since the more processed expressions are pretty opaque with out "internal" expressions in them.